### PR TITLE
Android: Modify touch screen GUI's buttons

### LINKED ISSUE #7219: Add in ability to turn off chat on mobile
**Issue type:**
 -  Feature request
 
**Summary:**
There are clients who play minetest on a mobile device (i.e. phone or tablet) and they currently don't have the ability to turn off chat for when they are recording a video (some servers require that they do this if they are going to record). Another reason to have the ability to turn off chat would be for phones that have a small screen size and large chunks of text said by a player hogging the screen and preventing them from being able to see what they are doing. I have played MT on a phone and having the chat hog the screen is a tad annoying while trying to build something as it blocks the view of what I am working on.

I suggest something like a little icon of a few lines in the same place as the other options under the little gear icon that is on the upper right side of the screen.

### PR DESCRIPTION
**Pull request by srifqi**
Add some buttons—listed below—to enhance playability.
- Zoom button is put above jump button.
- Minimap and toggle chat button are put in settings bar.

Move buttons on right screen a little.

Four new images:
(1) zoom (taken from minetest_game's `creative` mod),
(2) minimap,
(3) hide chat, and
(4) show chat.

Modified image:
(1) jump (uses rotated down).

Resolves #7219, #5173, and #2857.

new position of buttons

[Video (just showing feature, icons were not final)](https://youtu.be/Dry_hAnDkyc)

### CODE REVIEW

**Comment by srifqi** [comment]
Things I need suggestion about:
1. Should we hide some buttons when they can't be used, i.e. missing privileges? If yes, how it will respond to privileges change?
2. Even though it can be changed from Advanced Settings, should we add separate button to change round or square minimap?

**Comment by paramat** [comment]
1. Not sure, maybe later, always shown is simple and fine for now.
2. Seems not important enough, but it could be added later.

**Comment by paramat** [comment]
Video looks good. Could you also move the jump and down buttons upwards to be more usable? See https://github.com/minetest/minetest/issues/1750#issuecomment-380948675
Please could you use a vertically-flipped down / sneak arrow for jump / climb / fly up?

File: src/gui/touchscreengui.cpp [hunk 1]
```diff
@@ -519,6 +552,14 @@ void TouchScreenGUI::init(ISimpleTextureSource* tsrc)
 					m_screensize.Y),
 			L"H",false);
 
+	// init crunch button
+	initButton(zoom_id,
+			rect<s32>(m_screensize.X - (1.25 * button_size),
+					m_screensize.Y - (2.5 * button_size),
+					m_screensize.X - (0.25 * button_size),
+					m_screensize.Y - (1.5 * button_size)),
+			L"z",false);
```
Comment (SmallJoker) [inline src/gui/touchscreengui.cpp]: Missing space after `,`

File: src/gui/touchscreengui.cpp [hunk 2]
```diff
@@ -528,12 +569,17 @@ void TouchScreenGUI::init(ISimpleTextureSource* tsrc)
...
+	m_settingsbar.addButton(fly_id,     L"fly",       "fly_btn.png");
+	m_settingsbar.addButton(noclip_id,  L"noclip",    "noclip_btn.png");
+	m_settingsbar.addButton(fast_id,    L"fast",      "fast_btn.png");
+	m_settingsbar.addButton(debug_id,   L"debug",     "debug_btn.png");
+	m_settingsbar.addButton(camera_id,  L"camera",    "camera_btn.png");
+	m_settingsbar.addButton(range_id,   L"rangeview", "rangeview_btn.png");
+	m_settingsbar.addButton(minimap_id, L"minimap",   "minimap_btn.png");
+
+	// Chat is shown by default, so chat_hide_btn.png is shown first.
+	m_settingsbar.addToggleButton(toggle_chat_id,  L"togglechat",
```
Comment (SmallJoker) [inline src/gui/touchscreengui.cpp]: One superfluous space after `,`

**Comment by paramat** [comment]
Yes here https://github.com/minetest/minetest/blob/326eeca306f7bfb53ae3685eef18978dd81e587e/LICENSE.txt#L19

**Comment by paramat** [comment]
Nice thanks.

**Comment by srifqi** [comment]
Thank you for the merge!

Here is the APK from recent master branch just in case somebody wants to try.
APK (armeabi-v7a, 9877a1a207efd48b569cebe5c80e727477fb590f, minetest_game):
https://drive.google.com/open?id=1HOLrnKFnt7ym6O_gcJyAP4dXjhd0VvL7

**Comment by appgurueu** [comment]
Looks neat ! But doesn't it give Android players a major advantage(zooming) ? Especially on PVP Servers...

**Comment by srifqi** [comment]
@appgurueu No, it doesn't. Ability to zoom depends on each server's policy. Server owner can enable or disable zooming. This PR only add zoom button that is bound to zoom key (default: <kbd>Z</kbd>) on desktop.

**Comment by srifqi** [comment]
@appgurueu No, it doesn't. Ability to zoom depends on each server's policy. Server owner can enable or disable zooming. This PR only adds zoom button that is bound to zoom key (default: <kbd>Z</kbd>) on desktop.

**Pull request merged**